shell/platform/windows/window_unittests.cc

```diff
@@ -272,5 +284,78 @@ TEST(MockWindow, Paint) {
   window.InjectWindowMessage(WM_PAINT, 0, 0);
 }
 
+TEST(MockWindow, PointerHitTest) {
+  auto* windows_proc_table = new MockWindowsProcTable();
+  auto text_input_manager = std::make_unique<MockTextInputManager>();
+
+  MockWindow window(std::unique_ptr<MockWindowsProcTable>(windows_proc_table),
+                    std::move(text_input_manager));
+
+  auto direct_manipulation = new MockDirectManipulationOwner(&window);
+  window.SetDirectManipulationOwner(
+      std::unique_ptr<MockDirectManipulationOwner>(direct_manipulation));
+
+  UINT32 pointer_id = 123;
+
+  EXPECT_CALL(*windows_proc_table, GetPointerType(Eq(pointer_id), _))
+      .Times(1)
+      .WillOnce([](UINT32 pointer_id, POINTER_INPUT_TYPE* type) {
+        *type = PT_POINTER;
+        return TRUE;
+      });
+
+  EXPECT_CALL(*direct_manipulation, SetContact).Times(0);
+
+  window.InjectWindowMessage(DM_POINTERHITTEST, MAKEWPARAM(pointer_id, 0), 0);
+}
+
+TEST(MockWindow, TouchPadHitTest) {
+  auto* windows_proc_table = new MockWindowsProcTable();
+  auto text_input_manager = std::make_unique<MockTextInputManager>();
+
+  MockWindow window(std::unique_ptr<MockWindowsProcTable>(windows_proc_table),
+                    std::move(text_input_manager));
+
+  auto direct_manipulation = new MockDirectManipulationOwner(&window);
+  window.SetDirectManipulationOwner(
+      std::unique_ptr<MockDirectManipulationOwner>(direct_manipulation));
+
+  UINT32 pointer_id = 123;
+
+  EXPECT_CALL(*windows_proc_table, GetPointerType(Eq(pointer_id), _))
+      .Times(1)
+      .WillOnce([](UINT32 pointer_id, POINTER_INPUT_TYPE* type) {
+        *type = PT_TOUCHPAD;
+        return TRUE;
+      });
+
+  EXPECT_CALL(*direct_manipulation, SetContact(Eq(pointer_id))).Times(1);
+
+  window.InjectWindowMessage(DM_POINTERHITTEST, MAKEWPARAM(pointer_id, 0), 0);
+}
+
+TEST(MockWindow, UnknownPointerTypeSkipsDirectManipulation) {
+  auto* windows_proc_table = new MockWindowsProcTable();
```

Comment:
You could do the make unique here, then immediately, `EXPECT_CALL(table.get(), ...)` then move it into the window, which would avoid derefing a pointer you no longer own.

Reply:
Good idea, updated!